modules/apps/object/object-api/src/main/java/com/liferay/object/exception/ObjectValidationRuleSettingValueException.java

```diff
@@ -12,8 +12,48 @@
  */
 public class ObjectValidationRuleSettingValueException extends PortalException {
 
-	public ObjectValidationRuleSettingValueException(String msg) {
-		super(msg);
+	public ObjectValidationRuleSettingValueException(String message) {
+		super(message);
+
+		_messageKey = message;
+	}
+
+	public ObjectValidationRuleSettingValueException(
+		String message, String messageKey) {
+
+		super(message);
+
+		_messageKey = messageKey;
+	}
+
+	public String getMessageKey() {
+		return _messageKey;
+	}
+
+	public static class CompositeKeyMustHaveFiveObjectFieldsAtMaximum
```

Comment:
@mateussgn I prefer something more generic in case we change this max value in the future, something like `CompositeKeyMustHaveMaxObjectFields`